src/core/server/saved_objects/migrationsv2/index.ts

```diff
@@ -0,0 +1,857 @@
+        res.left.type === 'index_not_found_exception' &&
+        res.left.index === stateP.source.value
+      ) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          logs: [
+            ...stateP.logs,
+            {
+              level: 'error',
+              message: `LEGACY_REINDEX failed because the reindex destination index [${stateP.source.value}] does not exist.`,
+            },
+          ],
+        };
+      }
+      return { ...stateP, controlState: 'FATAL' };
+    }
+  } else if (stateP.controlState === 'LEGACY_DELETE') {
+    const res = resW as ResponseType<typeof stateP.controlState>;
+    if (Either.isLeft(res)) {
+      if (
+        res.left.type === 'remove_index_not_a_concrete_index' ||
+        (res.left.type === 'index_not_found_exception' && res.left.index === stateP.legacy)
+      ) {
+        // index_not_found_exception, another Kibana instance already
+        // deleted the legacy index
+        //
+        // remove_index_not_a_concrete_index, another Kibana instance already
+        // deleted the legacy index and created a .kibana alias
+        //
+        // If we detect that another instance has already completed a step, we
+        // can technically skip ahead in the process until after the completed
+        // step. However, by not skipping ahead we limit branches in the
+        // control state progression and simplify the implementation.
+        return { ...stateP, controlState: 'SET_SOURCE_WRITE_BLOCK' };
+      } else if (
+        res.left.type === 'index_not_found_exception' &&
+        res.left.index === stateP.source.value
+      ) {
+        return {
+          ...stateP,
+          controlState: 'FATAL',
+          logs: [
+            ...stateP.logs,
+            {
+              level: 'error',
+              message: `LEGACY_DELETE failed because the source index [${stateP.source.value}] does not exist.`,
+            },
+          ],
+        };
+      }
+    } else {
+      return { ...stateP, controlState: 'SET_SOURCE_WRITE_BLOCK' };
+    }
+    return stateP;
+  } else if (stateP.controlState === 'SET_SOURCE_WRITE_BLOCK') {
+    const res = resW as ResponseType<typeof stateP.controlState>;
+    // If the write block is sucessfully in place, proceed to the next step.
+    if (Either.isRight(res)) {
+      stateP = { ...stateP, controlState: 'CLONE_SOURCE_TO_TARGET' };
+    } else if (Either.isLeft(res)) {
+      if (res.left.type === 'index_not_found_exception') {
+        // If the write block failed because the index doesn't exist, it means
+        // another instance already completed the legacy pre-migration. Proceed
+        // to the next step.
+        stateP = { ...stateP, controlState: 'CLONE_SOURCE_TO_TARGET' };
+      }
+    } else {
+      return throwBadResponse(res);
+    }
+
+    return stateP;
+  } else if (stateP.controlState === 'CLONE_SOURCE_TO_TARGET') {
+    return { ...stateP, controlState: 'UPDATE_TARGET_MAPPINGS' };
```

Comment:
The clone task has either `Left<RetryableEsClientError>` or `Right`, so if the disk space is full it will throw an exception that will bubble up and crash the process.

Reply:
There's some discussion in https://github.com/elastic/kibana/pull/78413#discussion_r537112480 about if and how we want to handle such errors.